### PR TITLE
Corrupted Avar Chieftain / dusk rat change

### PR DESCRIPTION
**Pull request by GarOfMeridian**
You guys might remember this one. This one makes dusk rats no longer a morph option, while adding Corrupted Avar Chieftains that *can* be morphed into for the same poison.

Why?

* Dusk rats are, graphically, hilariously bad.

* Avar chieftains are, graphically, super freaking cool.

* If we're going to have players running around dealing poison with their bows, we should have them look cool and sensible while they do it.

* Because of their lower HP, dusk rats can be ill formed into under AMA and possibly even with Dement (I actually calculated this and used it in a tournament). Corrupted Chieftains will require a bit more spellpower to achieve this best illusionary form proc.

That's pretty much it. The Corrupted Avar Chieftain also spawns at Ruins of the Riija temple sometimes, and drops loot not normally found on the island (like DAFs). He also gives positive karma, another rare resource on the island. His theming is the same as shadowbeasts (kriipa + Qor, chieftain + Qor).

Obviously we lived without this for a year, but I also think our play experience and marketing and promotion materials will look cooler with Avar Chieftains bowing people down rather than pure black rat silhouettes clawing at the air like wtf. I guarantee nobody from outside Meridian has ever watched a PvP video and understood what the hell those rat silhouettes are doing all over.

### CODE REVIEW

File: kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+      oTreasure = Create(&NeruditeArrow,#corpse=corpse);
+      iNumber = Random(25,45);
+      Send(oTreasure,@AddNumber,#number=iNumber);
+
+      if Send(poOwner,@ReqNewHold,#what=oTreasure,
+               #new_row=piRow,#new_col=piCol)
+         AND Send(poOwner,@ReqSomethingMoved,#what=oTreasure,
+                  #new_row=piRow,#new_col=piCol)
+      {
+         Send(poOwner,@NewHold,#what=oTreasure,#new_row=piRow,#new_col=piCol);
```
Comment (skittles1) [inline kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod]: NewHold should have fine_row and fine_col parameters (see avchief.kod, this was fixed for all NewHold calls in the past few months and stops loot dropping inside walls).

File: kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+
+      if Random(1,POISON_CHANCE) = 1
+         AND IsClass(what,&Player)
+         AND NOT Send(what,@IsEnchanted,#byClass=&Poison)
+      {
+         oSpell = Send(SYS,@FindSpellByNum,#num=SID_POISON);
+         Send(oSpell,@MakePoisoned,#who=what,
+               #lossrate=POISON_LOSSRATE,#duration=POISON_DURATION);
+
+         if IsClass(what,&Player)
```
Comment (skittles1) [inline kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod]: Duplicate IsClass(what,&Player) check.

File: kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+            AND Send(what,@GetClan) <> Send(self,@GetClan)
+         {
+            return 25;
+         }
+      }
+
+      return 0;
+   }
+
+   GetTreasureType()
```
Comment (skittles1) [inline kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod]: If just returning the treasure type, you can delete this (monster.kod will return it).

**Comment by skittles1** [comment]
A possible issue with this is because mobs at Ruins kill each other, they don't drop loot. While this avar will, it might prevent players farming on that screen because of the lower payoff. Could stop the mobs fighting, or use a different screen. I probably wouldn't just make them all drop loot as they have a pretty good kill rate.

**Comment by GarOfMeridian** [comment]
I just removed them from spawns entirely for the time being. Their real purpose is for illusionary form. I have an idea for a jungle screen I can make later where their presence is appropriate.

File: kod/object/passive/trestype/avarchpoist.kod [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+constants:
+   
+   include blakston.khd
+   
+classvars:
+
+
+properties:
+   
+   piTreasure_num = TID_AVAR_CH_POIS
```
Comment (skittles1) [inline kod/object/passive/trestype/avarchpoist.kod]: Recent change in dev branch means piTreasure_num should be in classvars now, and called viTreasure_num

**Pull request closed without merging**